src/test/java/com/google/cloud/spanner/pgadapter/JdbcSimpleModeMockServerTest.java

```diff
@@ -259,6 +508,366 @@ public void testDdl() throws SQLException {
             .collect(Collectors.toList());
     assertEquals(1, updateDatabaseDdlRequests.size());
     assertEquals(1, updateDatabaseDdlRequests.get(0).getStatementsCount());
-    assertEquals(sql, updateDatabaseDdlRequests.get(0).getStatements(0));
+    assertEquals(
+        "CREATE TABLE foo (id bigint primary key)",
+        updateDatabaseDdlRequests.get(0).getStatements(0));
+
+    // Verify that the DML statements were batched together by PgAdapter.
+    List<ExecuteBatchDmlRequest> requests =
+        mockSpanner.getRequestsOfType(ExecuteBatchDmlRequest.class);
+    assertEquals(1, requests.size());
+    ExecuteBatchDmlRequest request = requests.get(0);
+    assertEquals(2, request.getStatementsCount());
+    assertEquals(INSERT_STATEMENT.getSql(), request.getStatements(0).getSql());
+    assertEquals(UPDATE_STATEMENT.getSql(), request.getStatements(1).getSql());
+  }
+
+  @Test
+  public void testErrorHandlingOfDdl() throws SQLException {
+    addDdlExceptionToSpannerAdmin();
+    String sql = String.format("%s; %s; %s", INSERT_STATEMENT, INVALID_DDL, UPDATE_STATEMENT);
+    try (Connection connection = DriverManager.getConnection(createUrl())) {
+      try (Statement statement = connection.createStatement()) {
+        SQLException exception = assertThrows(SQLException.class, () -> statement.execute(sql));
+        assertThat(
+            exception.getMessage(), containsString("INVALID_ARGUMENT: Statement is invalid."));
+      }
+    }
+
+    // Verify that the execution does not halt and the last statement is executed.
+    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+    assertEquals(2, requests.size());
+    assertEquals(INSERT_STATEMENT.getSql(), requests.get(0).getSql());
+    assertEquals(UPDATE_STATEMENT.getSql(), requests.get(1).getSql());
+  }
```

Comment:
nit: add an assertion that we received two `CommitRequest`s.

Reply:
Why two commit requests? Two is the correct answer but I can only think of one before INVALID_DDL.

Reply:
An invalid DDL statement should not cause the batch to be halted. So we get one commit request from the insert statement, then an invalid DDL statement, and then the update statement, which is also committed.

Reply:
Done.